tools/spectral/ipa/rulesets/functions/IPA109ValidOperationID.js

```diff
@@ -2,43 +2,67 @@ import { hasException } from './utils/exceptions.js';
 import { collectAdoption, collectException, collectAndReturnViolation } from './utils/collectionUtils.js';
 import { isCustomMethodIdentifier, getCustomMethodName, stripCustomMethodName } from './utils/resourceEvaluation.js';
 import { generateOperationID } from './utils/operationIdGeneration.js';
+import { hasMethodWithVerbOverride, hasCustomMethodOverride } from './utils/extensions.js';
 
 const RULE_NAME = 'xgen-IPA-109-valid-operation-id';
 const ERROR_MESSAGE = 'Invalid OperationID.';
 
 export default (input, _, { path }) => {
   let resourcePath = path[1];
-  const methodName = getCustomMethodName(resourcePath);
 
-  if (!isCustomMethodIdentifier(resourcePath)) {
+  if (!isCustomMethodIdentifier(resourcePath) && !hasMethodWithVerbOverride(input)) {
     return;
   }
 
-  // TODO detect custom method extension - CLOUDP-306294
+  let errors = [];
+  let expectedOperationID = '';
+  if (isCustomMethodIdentifier(resourcePath)) {
+    expectedOperationID = generateOperationID(getCustomMethodName(resourcePath), stripCustomMethodName(resourcePath));
```

Comment:
```suggestion
  if (isCustomMethodIdentifier(resourcePath)) {
    const expectedOperationID = generateOperationID(getCustomMethodName(resourcePath), stripCustomMethodName(resourcePath));
```
Since you are only using `expectedOperationID ` in the scope of each if block, you can declare it within. the block itself and assign

Reply:
Pushed a refactor, thank you!